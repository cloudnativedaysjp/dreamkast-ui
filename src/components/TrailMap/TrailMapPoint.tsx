import React from 'react'
import * as Styled from './styled'
import { useSelector } from 'react-redux'
import { settingsSelector } from '../../store/settings'

export const TrailMapPoint = () => {
  const settings = useSelector(settingsSelector)
  if (!settings.pointDataInitialized) {
    return <></>
  }

  const { total } = settings.pointData
  const ticket = Math.floor(total / 100)
  const frac = total % 100

  const trailingImage = (() => {
    if (frac < 20) {
      return (
        <img src={'/cndt2022/ui/trailmap_0.jpg'} alt={'Trail Mapの出発地点'} />
      )
    }
    if (frac < 40) {
      return (
        <img src={'/cndt2022/ui/trailmap_25.jpg'} alt={'Trail Mapの3合目'} />
      )
    }
    if (frac < 60) {
      return (
        <img src={'/cndt2022/ui/trailmap_50.jpg'} alt={'Trail Mapの5合目'} />
      )
    }
    if (frac < 80) {
      return (
        <img src={'/cndt2022/ui/trailmap_75.jpg'} alt={'Trail Mapの8合目'} />
      )
    }
    if (frac < 100) {
      return (
        <img
          src={'/cndt2022/ui/trailmap_100.jpg'}
          alt={'Trail Mapの山頂付近'}
        />
      )
    }
  })()

  return (
    <div>
      <Styled.TrailMapPointContainer>
        {trailingImage}
        <div className={'ticket'}>抽選券 {ticket} 枚ゲット！</div>
        <div className={'point'}>
          {ticket > 0 ? ticket * 100 : ''} + {frac}pt
        </div>
      </Styled.TrailMapPointContainer>
    </div>
  )
}
