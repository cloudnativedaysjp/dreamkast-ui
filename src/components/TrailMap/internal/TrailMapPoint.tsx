import React from 'react'
import * as Styled from '../styled'
import { Skeleton } from '@material-ui/lab'
import { ProfilePointsResponse } from '../../../generated/dreamkast-api.generated'

type Props = {
  isLoading: boolean
  pointData: ProfilePointsResponse
}

export const TrailMapPoint = ({ isLoading, pointData }: Props) => {
  if (isLoading) {
    return (
      <Styled.TrailMapPointSuspendContainer>
        <div className={'suspend'}>
          <Skeleton variant="rect" animation={'wave'} className={'skeleton'} />
        </div>
      </Styled.TrailMapPointSuspendContainer>
    )
  }

  const { total } = pointData
  const ticket = Math.floor(total / 100)
  const frac = total % 100

  const trailingImage = (() => {
    if (frac < 20) {
      return (
        <img src={'/cndt2022/ui/trailmap_0.jpg'} alt={'Trail Mapの出発地点'} />
      )
    }
    if (frac < 40) {
      return (
        <img src={'/cndt2022/ui/trailmap_25.jpg'} alt={'Trail Mapの3合目'} />
      )
    }
    if (frac < 60) {
      return (
        <img src={'/cndt2022/ui/trailmap_50.jpg'} alt={'Trail Mapの5合目'} />
      )
    }
    if (frac < 80) {
      return (
        <img src={'/cndt2022/ui/trailmap_75.jpg'} alt={'Trail Mapの8合目'} />
      )
    }
    if (frac < 100) {
      return (
        <img
          src={'/cndt2022/ui/trailmap_100.jpg'}
          alt={'Trail Mapの山頂付近'}
        />
      )
    }
  })()

  return (
    <Styled.TrailMapPointContainer>
      {trailingImage}
      {ticket > 0 ? (
        <div className={'ticket'}>抽選券 {ticket} 枚ゲット！</div>
      ) : (
        ''
      )}
      <div className={'point'}>
        {ticket > 0 ? `${ticket * 100} + ` : ''}
        {frac}pt
      </div>
    </Styled.TrailMapPointContainer>
  )
}
