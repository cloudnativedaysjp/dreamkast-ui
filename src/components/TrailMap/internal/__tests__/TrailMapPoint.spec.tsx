import React from 'react'
import { render } from '@testing-library/react'
import { TrailMapPoint } from '../TrailMapPoint'

describe('TrailMapPoint', () => {
  it('should render loading skeleton', () => {
    const data = {
      total: 20,
      points: [],
    }
    const screen = render(<TrailMapPoint pointData={data}></TrailMapPoint>)
    expect(screen.asFragment()).toMatchSnapshot()
  })

  it('should render loading skeleton when loading pointData', () => {
    const data = {
      total: 0,
      points: [],
    }
    const screen = render(
      <TrailMapPoint isLoading={true} pointData={data}></TrailMapPoint>,
    )
    expect(screen.asFragment()).toMatchSnapshot()
  })

  it('should render loading skeleton', () => {
    const data = {
      total: 0,
      points: [],
    }
    const screen = render(
      <TrailMapPoint isLoading={true} pointData={data}></TrailMapPoint>,
    )
    expect(screen.asFragment()).toMatchSnapshot()
  })
})
