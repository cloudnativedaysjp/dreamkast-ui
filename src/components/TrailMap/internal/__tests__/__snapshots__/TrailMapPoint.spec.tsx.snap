// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrailMapPoint should render loading skeleton 1`] = `
<DocumentFragment>
  <div
    class="styled__TrailMapPointSuspendContainer-sc-1t1auio-4 fhJyZv"
  >
    <div
      class="suspend"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-rect skeleton MuiSkeleton-wave"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`TrailMapPoint should render without crash 1`] = `
<DocumentFragment>
  <div
    class="styled__TrailMapPointContainer-sc-1t1auio-5 eZNYKX"
  >
    <img
      alt="Trail Mapの3合目"
      src="/cndt2022/ui/trailmap_25.jpg"
    />
    <div
      class="point"
    >
      20pt
    </div>
  </div>
</DocumentFragment>
`;
